keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,1084 @@
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+This change does not affect the API servers. This change may affect the ability
+of a kubelet to run a pod with gitRepo volumes.
+
+If there are no workloads using gitRepo volumes we don't envision any
+disruptions.
+
+If there are workloads using gitRepo volumes then kubelets that have been
+upgraded to have GitRepoVolumeDriver set to false will fail to run pods for
+these workloads.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Pods failing to start.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+Yes, we are disabling the in-tree gitRepo volume driver. We aren't removing it
+from the API.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Operators can check if there are pods with gitRepo volumes in their cluster by
+running the following command:
+
+```bash
+kubectl get pods -A -o json \
+| jq -r '.items[] | select(.spec.volumes[]? | has("hostPath")) | .metadata.namespace + " " + .metadata.name'
+```
+
+Operators can check if the feature-gate `GitRepoVolumeDriver` is enabled on a
+particular node by running the following command:
+
+```bash
+kubectl get --raw "/api/v1/nodes/<node-name>/proxy/metrics" \
+| grep kubernetes_feature_enabled \
+| grep GitRepoVolumeDriver
+```
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
```

Comment:
Should kubelet issue an event that it used an insecure volume?

Reply:
Is there precedent for doing that?

Reply:
I don't think so, but that doesn't make a bad idea out of hand :)

Reply:
Didn't mean to suggest that it was a bad idea, actually sounds like a good idea, I'm just not sure if its possible because I don't know enough about how those events are generated.

Reply:
I'm not saying it's a great idea either :) - it's an idea, please consider it and apply your context and wisdom.